src/utils/convert.js

```diff
@@ -585,15 +598,20 @@ const writeStep = (type, definition, key, legacyApp, newAppDir) => {
 };
 
 // render the authData used in the trigger/search/create test code
-const renderAuthData = (authType) => {
+const renderAuthData = (definition) => {
+  const authType = getAuthType(definition);
   let result;
   switch (authType) {
-  case 'basic':
+  case 'basic': {
+    let lines = _.map(definition.auth_fields, (field, key) => {
+      const upperKey = key.toUpperCase();
+      return `        ${key}: process.env.${upperKey}`;
+    });
     result = `{
-        username: process.env.USERNAME,
-        password: process.env.PASSWORD
+${lines.join(',\n')}
       }`;
     break;
+  }
```

Comment:
In the generated test code, we were assuming basic auth is always username and password, which is not true. We should really take advantage of `auth_fields` instead.